nixos/modules/services/web-apps/cryptpad.nix

```diff
@@ -0,0 +1,213 @@
+            type = types.str;
+            default = "info";
+            description = mdDoc "Controls log level";
+          };
+          blockDailyCheck = mkOption {
+            type = types.bool;
+            default = true;
+            description = mdDoc "Disable telemetry. This setting is only effective if the 'Disable server telemetry' setting in the admin menu has been untouched, and will be ignored once set there either way.";
+          };
+          installMethod = mkOption {
+            type = types.str;
+            default = "nixos";
+            description = mdDoc ''
+              Install method is listed in telemetry if you agree to it through the consentToContact
+              setting in the admin panel.
+            '';
+          };
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable (mkMerge [
+    {
+      users.users.cryptpad = {
+        isSystemUser = true;
+        group = "cryptpad";
+      };
+      users.groups.cryptpad = {};
+
+      systemd.services.cryptpad = {
+        description = "Cryptpad service";
+        wantedBy = [ "multi-user.target" ];
+        after = [ "networking.target" ];
+        serviceConfig = {
+          User = "cryptpad";
+          Environment = [
+            "CRYPTPAD_CONFIG=${cryptpadConfigFile}"
+            "HOME=%S/cryptpad"
+          ];
+          ExecStart = "${cfg.package}/bin/cryptpad";
+          PrivateTmp = true;
+          Restart = "always";
+          StateDirectory = "cryptpad";
+          WorkingDirectory = "%S/cryptpad";
+        };
+      };
+    }
+    (mkIf cfg.confinement {
+      systemd.services.cryptpad = {
+        serviceConfig.BindReadOnlyPaths = [
+          cryptpadConfigFile
+          # apparently needs proc for workers management
+          "/proc"
+          "/dev/urandom"
+        ] ++ (if ! cfg.config.blockDailyCheck then [
+          "/etc/resolv.conf"
+          "/etc/hosts"
+        ] else []);
+        confinement = {
+          enable = true;
+          binSh = null;
+          mode = "chroot-only";
+        };
+      };
+    })
+    (mkIf cfg.configureNginx {
+      assertions = [
+        { assertion = cfg.config.httpUnsafeOrigin != "";
+          message = "services.cryptpad.config.httpUnsafeOrigin is required";
+        }
+        { assertion = strings.hasPrefix "https://" cfg.config.httpUnsafeOrigin;
+          message = "services.cryptpad.config.httpUnsafeOrigin must start with https://";
+        }
+        { assertion = isNull cfg.config.httpSafeOrigin || strings.hasPrefix "https://" cfg.config.httpSafeOrigin;
+          message = "services.cryptpad.config.httpSafeOrigin must start with https:// (or be unset)";
+        }
+      ];
+      services.nginx = {
+        enable = true;
+        recommendedTlsSettings = true;
+
+        # FIXME: Check / compare this with [Nginx module configuration in nixpkgs](https://github.com/NixOS/nixpkgs/blob/nixos-23.11/nixos/modules/services/web-servers/nginx/default.nix).
```

Comment:
I've looked into this. [According to the Nginx doc](https://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_dhparam) DHE ciphers are disabled if ssl_dhparam is not set in Nginx versions after 1.11.0 (nixpkgs 23.11 currently uses Nginx 1.24.0).
Not having DHE ciphers enabled may cause some issues with older browsers, but I don't think that's something we need to worry about here so I suggest we safely remove this FIXME block.
Reference: https://security.stackexchange.com/questions/263040/tlsv1-should-dhe-be-enabled